src/django/web/templates/web/environment.js

```diff
@@ -0,0 +1 @@
+window.ENVIRONMENT = Object.freeze({{ environment | safe }});
```

Comment:
For my own learning can you walk me through this line? I think I understand this as making the `environment` property immutable which we use as an environment var to construct the right tiles url, but I'm not sure where `safe` comes from.

Reply:
This file is really a Jinja template, used by Django to render a page. Usually we use Jinja syntax in HTML files, and it's more obvious. Here we're using it in a JavaScript file.

The template is populated with values here:

https://github.com/azavea/iow-boundary-tool/blob/ce02f5e9a5930ce7a3ed4b6920f1712946e3d87d/src/django/web/views.py#L17-L24

The `environment` in the template is populated using the `context` that is provided to the `render` function.

The `safe` is a Jinja filter, used to escape HTML so that the output is not interpreted and executed by the browser, but is taken literally. This changes `<` into `&lt;`, `&` into `&amp;`, etc. For more detail, see here: https://jinja.palletsprojects.com/en/3.1.x/templates/#html-escaping

Reply:
Oh! Thank you for explaining that makes sense!